### DISCUSSION TITLE
전예린

### DISCUSSION THREAD

**Discussion by z1nun** [z1nun/ASTERISK-#20] [독서일지]
# 각자 선정한 개발자 필독서를 읽고 그날 읽은 내용을 간단히 정리해서 적어주세요!

## 커리어 스킬 완벽한 개발자 인생 로드맵

- 읽은 내용은 밑에 comments에 적어주세요!

**Comment by sweetyr928** [comment]
22.10.02 일요일

31p~66p

어떤 계획을 세워야할지, 어떤 개발자가 되고 싶은지 먼저 생각해라.
프로그래밍 언어 한 가지를 익혀라. 어떤 언어를 배울지는 생각만큼 중요치않다.
코드를 구조화 해라. 코딩을 처음 배울 때부터 깔끔하게 잘 작성하는 방법을 배워라(코드만으로도 의사 전달이 가능하게끔).
알고리즘을 다양하게 변형하는 방법을 익혀라(알고리즘은 배우기 어렵지만 가치가 있음).
개발 플랫폼 하나와 그와 함께 사용하는 기술 또는 프레임워크를 익숙하게 쓸 수 있어야한다. 선택한 프레임워크에 따른 전체 개발 스택을 배워라(스텍은 애플리케이션 하나를 제대로 완성하는데 함께 쓰이는 기술 세트를 말함).

**Comment by sweetyr928** [comment]
22.10.03 월요일

67p~86p

어떤 기술이든 발전시키려면 시간이 든다. 연습해라. 시간이 오래걸리고 발전이 정체되고 있다고 느낄 때를 조심해라. 확실한 계획과 명확한 목표를 따라가는 한 기술은 분명 발전할 것이다.
첫번째,두번째 프로그래밍 언어를 배울 때가 가장 어렵다. 프로그래밍 언어들은 공통점이 많으니 한 두개를 배워두면 점차 수월해질 것이다.
지금은 라이브러리와 프레임워크를 어떻게 활용하느냐가 프로그래밍 언어를 깊게 아는 것보다 중요하다. 그러니 배우겠다고 선택한 언어가 어려워도 포기하지마라.

**Comment by sweetyr928** [comment]
22.10.04 화요일

86p~109p

<첫번째 프로그래밍 언어 배우기>
1. 잘 작동하는 애플리케이션 살펴보기
프로그래밍을 배울 때 책을 찾는 것도 좋지만 잘 작동하는 애플리케이션의 소스 코드를 살펴보고(이왕이면 인기 있는 것. 설계가 잘 되어있을 가능성이 높기 때문) 코드가 어떤 역할을 하는지 느껴라. 
2. 몇 가지 참고자료 훑어보기(책,동영상,기사 튜토리얼 등) 
프로그래밍 관련 도서를 처음부터 끝까지 꼼꼼히 읽지 않아도 된다. 자신이 배우려는 언어의 범위, 기본 개념은 무엇인지 정도만 이해하자.
3. 'Hello world' 프로그램 만들기
해당 언어를 제대로 배운다기보다 그 언어로 프로그램을 만들고 실행하는 데 필요한 기본 툴 체인을 익히고 테스트해보자.
4. 기본 문법을 배우고 실제 문제로 테스트해보기
5. 기능과 라이브러리 구분하기
특정 언어에서 언어와 라이브러리의 영역이 어떻게 나뉘는지 이해하고, 그 언어로 문제를 해결하기 위해 라이브러리를 찾는 법을 익혀라.
6. 기존 코드 리뷰하고 한 줄씩 이해하기
7. 무엇이든 만들어보기
소소한 프로젝트 아이디어를 몇 개 골라 애플리케이션을 만들어라. 자신이 배운 프로그래밍 언어와 표준 라이브러리를 활용하는데 중점을 둔 단순한 애플리케이션을 만드는데 집중해라.
8. 프로그래밍 언어를 특정 기술이나 플랫폼에 적용해보기
ex) 자바를 배우는 중이라면 안드로이드 프레임워크와 앱 작성 법 배운 후 안드로이드 앱 제작해보기
9. 어려운 알고리즘 문제 해결해보기
내가 풀지 못하고 끙끙대는 문제를 다른 사람이 어떻게 풀었는지 반드시 살펴볼 것. 처음엔 어려운게 정상이다. 시간이 지날수록 문제 유형에 익숙해지고 해결책도 즉시 떠오른다.

**Comment by sweetyr928** [comment]
22.10.05 수요일

110p~134p

<프로그래밍 독학 전략>

1. 무엇을 공부할지 계획을 세워라. 얼마의 기간을 들일지, 목표로 하는 수준에 이르기까지 구체적으로 어떤 단계를 거칠지도 계획하라. 
2. 일정표를 만들어라. 일정을 미리 정해두고 마치 거기에 목숨이 걸린 것처럼 지켜라. 
3. 공부하는 동안 인맥을 쌓아라. 
4. 멘토를 찾아라. 소프트웨어 개발자 한 명 정도는 알고 지내는 게 좋다. 질문이 생길 때 물어보거나 꼭 필요할 때 도움을 청할 수 있는 사람 말이다.
5. 사이드 프로젝트를 진행해라. 좋은 사이드 프로젝트는 다음과 같은 혜택을 제공한다.
      - 배운 내용을 연습하고 적용해볼 문제 제공
      - 포트폴리오 제작에 보탬이 됨
      - 추가 소득이나 새로운 사업을 만들 가능성이 열림
      - 코딩을 배우는 동시에 유용한 도구도 만들 수 있음
      - 자신의 능력에 대한 자신감 키워줌
      - (이런 일을 좋아하는 사람에게는) 그 자체로 재미있음
6. 플루럴 사이트 같은 곳에 가입하라(린다, 유데미, 트리우스 등과 비슷한). 혼자 체계적으로 공부할 수 있는 아주 좋은 방법이다.

**Comment by sweetyr928** [comment]
22.10.06 목요일

135p~159p

<인턴십>

거대 IT 회사는 보통 인턴이나 경력직 개발자만 고용한다. 해볼 기회가 생겼다면, 특히 막 입문할 즈음 그런 기회를 얻었다면 보수에 상관없이 그 기회를 꼭 잡아라. 인턴이되면 자신이 무슨 일을 해야 할지 스스로 알아내고 최소의 관리와 피드백만 받고도 일을 제대로 해야 한다. 다른 직원이 더 수월하게 일할 수 있게 노력해야 한다는 뜻이다.

<경력없이 일자리 구하기>

경력 없이 소프트웨어 개발자로 취직할 핵심 전략은 자신이 실제 코딩 능력을 갖추었다는 사실을 증명하는 것이다. 아직 업무 경력이 없어서 그렇지 프로그래밍 경력은 충분히 쌓았다는 사실을 확실히 보여주고 코딩 능력을 입증할 증거를 제시하면 된다. 
온라인에 존재감을 드러내라. 소프트웨어 개발자라면 블로그가 있어야 한다. 
작성한 코드와 참여한 프로젝트를 포트폴리오에 잘 녹여내라. 깃허브는 직장 경력이 부족할 때 자신의 능력을 증명할 수 있는 방법 중 하나다.포트폴리오에 최소 서너 가지 이상의 애플리케이션이나 프로젝트를 넣어라. 대단할 필요는 없지만 그렇다고 너무 사소한 건 곤란하다. 단위 테스트나 자동테스트도 포함해 테스트 코드를 작성할 능력이 있다는 것도 증명해라. 
자격증만으로 그 분야에 대한 지식을 증명하거나 일자리를 보장받기는 어렵다. 하지만 경력이 부족한 사람을 고용하며 느끼는 불안감은 경감시킬 수 있다.
경력이 부족하다면 진취적인 태도로 이를 보충해야 한다. 상대에게 긍정적인 답을 얻을 때까지 끈질기게 매달리는 진취적인 사람이 돼라.
